clang/include/clang/3C/ConstraintVariables.h

```diff
@@ -319,7 +319,10 @@ class PointerVariableConstraint : public ConstraintVariable {
   // Generic types can be used with fewer restrictions, so this field is used
   // stop assignments with generic variables from forcing constraint variables
   // to be wild.
-  int GenericIndex;
+  //
+  // Base is generated from the source code, New is set internally
+  int BaseGenericIndex;
+  int NewGenericIndex;
```

Comment:
Do these variables serve the same purpose, and the difference is that `Base` is due to what's in the code already, and `New` is what you might have inferred? Is it possible to (in the future, with a full design) retain something in the code, and also infer new generic parameters?

Reply:
Mostly I needed a way to indicate a change, and I expect them to be equal for most things. Currently, Base would be -1 (not generic) and New would be 0 (first generic param) when we need to add generics annotations. For the future, as you say, they could be used to alter existing params. I don't know when we'd have a reason for that, but maybe it will be easier to dole out params in some order.

Reply:
I was thinking that you might have
```
_Itype_for_any(T) int foo(void *x : itype(_Ptr<T>), void *y) { y = x; return 0; }
```
Here one generic parameter is given, and the other is not, but could be inferred.